gptel.el

```diff
@@ -780,7 +797,33 @@ file."
 ;; Minor mode and UI
 
 ;; NOTE: It's not clear that this is the best strategy:
-(add-to-list 'text-property-default-nonsticky '(gptel . t))
```

Comment:
I believe this should be the only point of contention.

Reply:
As far as the rest of this PR goes, I feel the text properties work much better as the role tagging mechanism for the chat buffer than overlays, given the downsides discussed in https://github.com/karthink/gptel/discussions/321.